### PR TITLE
[stable/prometheus-operator] NetworkPolicy support

### PR DESCRIPTION
**Pull request by antongorkovenko**
Implements support of Network Policies for Prometheus Operator

Signed-off-by: Anton Gorkovenko <gostom@gmail.com>

#### Is this a new chart
no
#### What this PR does / why we need it:
Implements support of Network Policies for Prometheus Operator
#### Which issue this PR fixes
no
#### Special notes for your reviewer:
no
#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by bismarck** [comment]
@antongorkovenko Thanks for the PR.  This might take a little time to review because network policies are a bit tricky.  What CNI plugin did you test these changes with?

**Comment by antongorkovenko** [comment]
@bismarck Calico

File: stable/prometheus-operator/templates/alertmanager/networkPolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      app: {{ template "prometheus-operator.name" . }}-alertmanager
+{{ include "prometheus-operator.labels" . | indent 6 }}
+  ingress:
+    - from:
+        - podSelector:
+            matchLabels:
+              app: {{ template "prometheus-operator.name" . }}-prometheus
+{{ include "prometheus-operator.labels" . | indent 14 }}
+    - ports:
+        - port: 9093
```
Comment (vsliouniaev) [inline stable/prometheus-operator/templates/alertmanager/networkPolicy.yaml]: I believe this port is templated

**Review by vsliouniaev — CHANGES_REQUESTED** [review]

File: stable/prometheus-operator/templates/alertmanager/networkPolicy.yaml (hunk 1 above)
Comment (antongorkovenko) [reply stable/prometheus-operator/templates/alertmanager/networkPolicy.yaml]: fixed

File: stable/prometheus-operator/templates/prometheus/networkPolicy.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    app: {{ template "prometheus-operator.name" . }}-prometheus
+{{ include "prometheus-operator.labels" . | indent 4 }}
+spec:
+  podSelector:
+    matchLabels:
+      app: {{ template "prometheus-operator.name" . }}-prometheus
+{{ include "prometheus-operator.labels" . | indent 6 }}
+  ingress:
+    - ports:
+      - port: 9090
```
Comment (bismarck) [inline stable/prometheus-operator/templates/prometheus/networkPolicy.yaml]: This is also templated out.

**Comment by bismarck** [comment]
@antongorkovenko I think some of these NetworkPolicies are being installed in the wrong namespace.  For example, kube-controller-manager is in the kube-system namespace which isn't necessarily the same namespace that Prometheus is deployed to. In that case you may also need to [add a namespace selector (in addition to your pod selector)](https://github.com/ahmetb/kubernetes-network-policy-recipes/blob/master/07-allow-traffic-from-some-pods-in-another-namespace.md) to your network policy, which introduces other complications.

**Review by bismarck — CHANGES_REQUESTED** [review]

File: stable/prometheus-operator/templates/prometheus/networkPolicy.yaml (hunk 2 above)
Comment (antongorkovenko) [reply stable/prometheus-operator/templates/prometheus/networkPolicy.yaml]: @bismarck updated

**Comment by antongorkovenko** [comment]
> @antongorkovenko I think some of these NetworkPolicies are being installed in the wrong namespace. For example, kube-controller-manager is in the kube-system namespace which isn't necessarily the same namespace that Prometheus is deployed to. In that case you may also need to [add a namespace selector (in addition to your pod selector)](https://github.com/ahmetb/kubernetes-network-policy-recipes/blob/master/07-allow-traffic-from-some-pods-in-another-namespace.md) to your network policy, which introduces other complications.

In progress

**Pull request closed without merging**